### PR TITLE
Change q30 threshold values on reads aggregation page

### PR DESCRIPTION
**Pull request by sofiahag**
As requested by Pär.

I cleaned up the python code a bit, hope that's ok.

I added "mode" and "threshold" values to the table, not sure if these are needed? I added them to easier derive the q30 colors.

I just added "q30>threshold" and "q30<threshold" metrics now. Do we want another limit value?

I'll deploy in stage. To see a project with q30 values below threshold, check P30702.

I'll also ask Pär if he's happy with it!

### CODE REVIEW

**Comment by sofiahag** [comment]
Up on stage! I stole it from Johannes 😈 maybe he'll steal it back though. I saw that the test project isn't available on stage, so maybe we need to update the database to see some projects.

File: run_dir/design/reads_total.html [hunk 1]
```diff
@@ -57,25 +57,55 @@ <h3 id="summary_table_section">Summary of Selected {% if "isHiseqX" in readsdata
...
+                            {% else %}
+                                {% if run_setup >= 250 %}
+                                    {% set q30_threshold = 60.0 %}
+                                {% elif run_setup >= 150 %}
+                                    {% set q30_threshold = 75.0 %}
+                                {% elif run_setup >= 100 %}
+                                    {% set q30_threshold = 80.0 %}
+                                {% elif run_setup < 100 %}
+                                    {% set q30_threshold = 85.0 %}
+                                {% end %}
```
Comment (aanil) [inline run_dir/design/reads_total.html:92]: Move the threshold definition outside the loop perhaps?

File: status/flowcells.py [hunk 2]
```diff
@@ -495,48 +495,63 @@ class ReadsTotalHandler(SafeHandler):
...
+                                        })
+                                        break
+
+        return data
+
+    def order_data(self, data):
+        ordered_data = OrderedDict()
+        for key in sorted(data.keys()):
+            ordered_data[key] = sorted(data[key], key=lambda d: d["fcp"])
+        return ordered_data
```
Comment (aanil) [inline status/flowcells.py:554]: Can't this be part of the retrieve_data function? Its only ever called there

**Review by parlundin — APPROVED** [review]
Tack!

File: status/flowcells.py (hunk 2 above)
Comment (sofiahag) [reply status/flowcells.py:554]: Ok, I put back the old structure instead 👍🏻

**Comment by sofiahag** [comment]
Now the Highcharts columns for data > threshold are yellow 🙄 no logical explanation for that, I haven't even changed that code 🤯 I'll look at it!

**Comment by sofiahag** [comment]
See difference between https://genomics-status.scilifelab.se/reads_total/P30059 and https://genomics-status-stage.scilifelab.se/reads_total/P30059

File: status/flowcells.py (hunk 2 above)
Comment (aanil) [reply status/flowcells.py:554]: I don't see it in the PR, maybe it wasn't pushed?

File: run_dir/design/reads_total.html (hunk 1 above)
Comment (aanil) [reply run_dir/design/reads_total.html]: I changed it a bit now!

File: status/flowcells.py (hunk 2 above)
Comment (sofiahag) [reply status/flowcells.py]: I didn't even commit the changes! 😆

File: run_dir/design/reads_total.html (hunk 1 above)
Comment (sofiahag) [reply run_dir/design/reads_total.html]: Ok, that looks cleaner!

**Comment by sofiahag** [comment]
Up on stage!

File: status/flowcells.py [hunk 3]
```diff
@@ -527,6 +527,26 @@ def get(self, query):
                         if row.key[1] + ":" + row.key[2] == fcl["fcp"]:
                             fcl["sample_status"] = row.value["sample_status"]
                             break  # since the row is already found
+            #To add correct threshold values
+            for row in fc_view:
+                for x, y in row.items():
+                    for lane, dic in row.value["lane_info"].items():
+                        for key, value in data.items():
+                            for dicti in value:
```
Comment (aanil) [inline status/flowcells.py:535]: What does dic and dicti stand for? More descriptive names would be better.

**Review by aanil — APPROVED** [review]
👍

**Pull request merged**